### DISCUSSION TITLE
Is there a yuck reference?

### DISCUSSION THREAD

**Discussion by akiross** [elkowar/eww#806] [Q&A]
Hello! First of all, thanks for eww, which I just started exploring and it's pretty cool! :)

Then, my question: while I'm familiar with functional programming, I'm not familiar with yuck and it seems I'm unable to get a few things right. To gather info I have to search in examples, but it's pretty inefficient and having a (short) reference to the language would help much.

Fore example, searching in the docs there's no occurrence of the word "comment", but comments are present somewhere in the code. Other random questions that pop in my mind atm are:
- are multi-line comments available? (Yes, but I only found them mentioned in a discussion here)
- are let-expressions available?
- what are the supported data types/are there specific assumptions made about type (and their casts)?
- what functions are available (e.g. string manipulation ones)?

I think having an answer to this kind of questions might help people getting started; is there some kind of reference like this that I missed?

Side question: out of curiosity, why isn't eww using a general purpose configuration language, instead of it's own? Given yuck is functional, handles json and the project uses rust and nix, I think [nickel](https://nickel-lang.org/) could be a good fit to configure eww.

**Comment by mantacid** [comment]
I don’t have the answers to any of these questions, but I too find the current documentation to be lacking in some of the more intricate aspects of the language.

**Reply by mantacid** [reply]
So there is a yuck reference, but it’s just the official docs, the unofficial guide by dharmx, and whatever stuff you can reverse engineer from people’s dotfiles.
